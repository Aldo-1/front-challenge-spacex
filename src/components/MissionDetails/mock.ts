export default {
  mission_name: 'Starlink-15',
  details:
    'SpaceX will launch Sentinel-6 Michael Freilich into low Earth orbit for NASA, NOAA, ESA, and the European Organization for the Exploitation of Meteorological Satellites aboard a Falcon 9 from SLC-4E, Vandenberg Air Force Station. Sentinel-6(A) is an ocean observation satellite providing radar ocean surface altimetry data and also atmospheric temperature profiles as a secondary mission. The booster for this mission is will land at LZ-4.',
  ships: [
    {
      name: 'GO Ms Tree',
      home_port: 'Port Canaveral',
      image: 'https://i.imgur.com/MtEgYbY.jpg'
    },
    {
      name: 'GO Ms Chief',
      home_port: 'Port Canaveral',
      image: 'https://imgur.com/NHsx95l.jpg'
    },
    {
      name: 'Just Read The Instructions 2',
      home_port: 'Port of Los Angeles',
      image: 'https://i.imgur.com/7VMC0Gn.jpg'
    },
    {
      name: 'GO Quest',
      home_port: 'Port Canaveral',
      image: 'https://i.imgur.com/ABXtHKa.jpg'
    }
  ],
  links: {
    article_link: 'https://youtu.be/J442-ti-Dhg',
    video_link: 'https://youtu.be/J442-ti-Dhg'
  }
}
