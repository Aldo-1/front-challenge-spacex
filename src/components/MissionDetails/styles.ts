import styled, { css } from 'styled-components'

const baseTitleStyles = css`
  color: #273944;
  margin-bottom: 1.2rem;
  text-align: center;
`

export const Wrapper = styled.main`
  border: 2px solid #c2d0d8;
  display: flex;
  align-items: center;
  background-color: #ffff;
  flex-direction: column;
`

export const Image = styled.img`
  width: 100%;
`

export const Title = styled.h1`
  font-size: 2.4rem;
  ${baseTitleStyles}
`
export const BackIcon = styled.a`
  position: relative;
  display: block;
  width: 100%;
  svg {
    position: absolute;
    left: 10px;
    top: 10px;
    max-width: 30px;
    color: white;
  }
`
export const Description = styled.p`
  font-size: 1.4rem;
  ${baseTitleStyles}
  margin-bottom: 0px;
`

export const Ships = styled.ul`
  align-self: flex-start;
`
export const Content = styled.div`
  padding: 12px 14px;
  display: flex;
  flex-direction: column;
  width: 100%;
`
export const Ship = styled.li`
  list-style: none;
  display: flex;
  align-items: center;
  span {
    font-size: 1.4rem;
    color: #273944;
    margin-left: 1.2rem;
  }
  margin-top: 1.2rem;
`
export const Subtitle = styled.h2`
  ${baseTitleStyles}
  text-align: start;
  font-size: 1.8rem;
  margin-bottom: 0px;
  margin-top: 1.2rem;
`
export const Thumb = styled.img`
  width: 100px;
  border-radius: 6px;
  height: 73px;
  object-fit: cover;
`
export const WrapperLinks = styled.div`
  align-self: center;
  padding-top: 1.6rem;
  padding-bottom: 1.6rem;
  margin-top: 1.2rem;
`

export const Link = styled.a`
  text-decoration: none;
  background-color: #c2d0d8;
  color: #273944;
  font-size: 1.4rem;
  border-radius: 6px;
  padding: 1.6rem;
  &:first-child {
    margin-right: 1.2rem;
  }
`
