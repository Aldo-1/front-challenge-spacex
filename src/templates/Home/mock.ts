export default [
  {
    mission_name: 'Starlink-15 (v1.0)',
    mission_link: 'starlink-15',
    launch_date: '2020-10-24T11:31:00-04:00'
  },
  {
    mission_name: 'Starlink-15 (v1.0)',
    mission_link: 'starlink-15',
    launch_date: '2020-10-24T11:31:00-04:00'
  },
  {
    mission_name: 'Starlink-15 (v1.0)',
    mission_link: 'starlink-15',
    launch_date: '2020-10-24T11:31:00-04:00'
  },
  {
    mission_name: 'Starlink-15 (v1.0)',
    mission_link: 'starlink-15',
    launch_date: '2020-10-24T11:31:00-04:00'
  },
  {
    mission_name: 'Starlink-15 (v1.0)',
    mission_link: 'starlink-15',
    launch_date: '2020-10-24T11:31:00-04:00'
  },
  {
    mission_name: 'Starlink-15 (v1.0)',
    mission_link: 'starlink-15',
    launch_date: '2020-10-24T11:31:00-04:00'
  },
  {
    mission_name: 'Starlink-15 (v1.0)',
    mission_link: 'starlink-15',
    launch_date: '2020-10-24T11:31:00-04:00'
  },
  {
    mission_name: 'Starlink-15 (v1.0)',
    mission_link: 'starlink-15',
    launch_date: '2020-10-24T11:31:00-04:00'
  },
  {
    mission_name: 'Starlink-15 (v1.0)',
    mission_link: 'starlink-15',
    launch_date: '2020-10-24T11:31:00-04:00'
  },
  {
    mission_name: 'Starlink-15 (v1.0)',
    mission_link: 'starlink-15',
    launch_date: '2020-10-24T11:31:00-04:00'
  },
  {
    mission_name: 'Starlink-15 (v1.0)',
    mission_link: 'starlink-15',
    launch_date: '2020-10-24T11:31:00-04:00'
  }
]
