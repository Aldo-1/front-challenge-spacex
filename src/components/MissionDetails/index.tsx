import * as S from './styles'

type ShipsProps = {
  name: string
  home_port: string
  image: string
}
type LinksProps = {
  article_link: string | null
  video_link: string | null
}
export type MissionDetailsProps = {
  mission_name: string
  details: string | null
  ships: ShipsProps[]
  links: LinksProps
}

const MissionDetails = ({
  mission_name,
  details,
  ships,
  links
}: MissionDetailsProps) => (
  <S.Wrapper>
    <S.Image src="img/rocket.jpg" alt="Rocket image"></S.Image>
    <S.Content>
      <S.Title>{mission_name}</S.Title>
      {details && <S.Description>{details}</S.Description>}
      <S.Subtitle>Ships:</S.Subtitle>
      <S.Ships aria-label="ships">
        {ships?.map((item, id) => (
          <S.Ship key={id}>
            <S.Thumb alt="Thumb image" src={item.image}></S.Thumb>
            <span>
              {item.name}
              <br />
              Home port: {item.home_port}
            </span>
          </S.Ship>
        ))}
      </S.Ships>
      <S.WrapperLinks>
        {links.article_link && (
          <S.Link
            href={links.article_link}
            title="Article Link"
            target="_blank"
          >
            Article Link
          </S.Link>
        )}
        {links.video_link && (
          <S.Link title="Video Link" href={links.video_link} target="_blank">
            Video Link
          </S.Link>
        )}
      </S.WrapperLinks>
    </S.Content>
  </S.Wrapper>
)

export default MissionDetails
