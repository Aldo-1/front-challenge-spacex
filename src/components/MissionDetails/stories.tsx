import { Story, Meta } from '@storybook/react/types-6-0'
import MissionDetails, { MissionDetailsProps } from '.'
import { Container } from 'components/Container'
import mock from './mock'

export default {
  title: 'MissionDetails',
  component: MissionDetails
} as Meta

export const Default: Story<MissionDetailsProps> = (args) => (
  <Container>
    <MissionDetails {...args} />
  </Container>
)

Default.args = mock
