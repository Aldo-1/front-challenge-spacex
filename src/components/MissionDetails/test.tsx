import { render, screen, within } from '@testing-library/react'

import MissionDetails from '.'
import mock from './mock'

describe('<MissionDetails />', () => {
  it('should render the heading, the details, the ships and the links', () => {
    render(<MissionDetails {...mock} />)
    expect(
      screen.getByRole('heading', { name: 'Starlink-15' })
    ).toBeInTheDocument()

    expect(
      screen.getByText(
        'SpaceX will launch Sentinel-6 Michael Freilich into low Earth orbit for NASA, NOAA, ESA, and the European Organization for the Exploitation of Meteorological Satellites aboard a Falcon 9 from SLC-4E, Vandenberg Air Force Station. Sentinel-6(A) is an ocean observation satellite providing radar ocean surface altimetry data and also atmospheric temperature profiles as a secondary mission. The booster for this mission is will land at LZ-4.'
      )
    ).toBeInTheDocument()

    const list = screen.getByRole('list', { name: /ships/i })
    const { getAllByRole } = within(list)
    const items = getAllByRole('listitem')
    expect(items.length).toBe(4)

    expect(
      screen.getByRole('link', { name: 'Article Link' })
    ).toBeInTheDocument()

    expect(screen.getByRole('link', { name: 'Video Link' })).toBeInTheDocument()
  })

  it('should render without description', () => {
    render(<MissionDetails {...mock} details={null} />)
    expect(screen.queryByRole('paragraph')).not.toBeInTheDocument()
  })

  it('should render without article link', () => {
    const links = {
      article_link: null,
      video_link: 'https://youtu.be/J442-ti-Dhg'
    }
    render(<MissionDetails {...mock} links={links} />)
    expect(
      screen.queryByRole('link', { name: 'Article Link' })
    ).not.toBeInTheDocument()

    expect(screen.getByRole('link', { name: 'Video Link' })).toBeInTheDocument()
  })

  it('should render without video link', () => {
    const links = {
      article_link: 'https://youtu.be/J442-ti-Dhg',
      video_link: null
    }
    const { container } = render(<MissionDetails {...mock} links={links} />)
    expect(
      screen.queryByRole('link', { name: 'Video Link' })
    ).not.toBeInTheDocument()

    expect(
      screen.getByRole('link', { name: 'Article Link' })
    ).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })
})
