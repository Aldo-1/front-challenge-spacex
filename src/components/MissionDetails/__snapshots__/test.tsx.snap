// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MissionDetails /> should render without video link 1`] = `
.c0 {
  border: 2px solid #c2d0d8;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #ffff;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c1 {
  width: 100%;
}

.c3 {
  font-size: 2.4rem;
  color: #273944;
  margin-bottom: 1.2rem;
  text-align: center;
}

.c4 {
  font-size: 1.4rem;
  color: #273944;
  margin-bottom: 1.2rem;
  text-align: center;
  margin-bottom: 0px;
}

.c6 {
  -webkit-align-self: flex-start;
  -ms-flex-item-align: start;
  align-self: flex-start;
}

.c2 {
  padding: 12px 14px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c7 {
  list-style: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-top: 1.2rem;
}

.c7 span {
  font-size: 1.4rem;
  color: #273944;
  margin-left: 1.2rem;
}

.c5 {
  color: #273944;
  margin-bottom: 1.2rem;
  text-align: center;
  text-align: start;
  font-size: 1.8rem;
  margin-bottom: 0px;
  margin-top: 1.2rem;
}

.c8 {
  width: 100px;
  border-radius: 6px;
  height: 73px;
  object-fit: cover;
}

.c9 {
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  padding-top: 1.6rem;
  padding-bottom: 1.6rem;
  margin-top: 1.2rem;
}

.c10 {
  -webkit-text-decoration: none;
  text-decoration: none;
  background-color: #c2d0d8;
  color: #273944;
  font-size: 1.4rem;
  border-radius: 6px;
  padding: 1.6rem;
}

.c10:first-child {
  margin-right: 1.2rem;
}

<main
  class="c0"
>
  <img
    alt="Rocket image"
    class="c1"
    src="img/rocket.jpg"
  />
  <div
    class="c2"
  >
    <h1
      class="c3"
    >
      Starlink-15
    </h1>
    <p
      class="c4"
    >
      SpaceX will launch Sentinel-6 Michael Freilich into low Earth orbit for NASA, NOAA, ESA, and the European Organization for the Exploitation of Meteorological Satellites aboard a Falcon 9 from SLC-4E, Vandenberg Air Force Station. Sentinel-6(A) is an ocean observation satellite providing radar ocean surface altimetry data and also atmospheric temperature profiles as a secondary mission. The booster for this mission is will land at LZ-4.
    </p>
    <h2
      class="c5"
    >
      Ships:
    </h2>
    <ul
      aria-label="ships"
      class="c6"
    >
      <li
        class="c7"
      >
        <img
          alt="Thumb image"
          class="c8"
          src="https://i.imgur.com/MtEgYbY.jpg"
        />
        <span>
          GO Ms Tree
          <br />
          Home port: 
          Port Canaveral
        </span>
      </li>
      <li
        class="c7"
      >
        <img
          alt="Thumb image"
          class="c8"
          src="https://imgur.com/NHsx95l.jpg"
        />
        <span>
          GO Ms Chief
          <br />
          Home port: 
          Port Canaveral
        </span>
      </li>
      <li
        class="c7"
      >
        <img
          alt="Thumb image"
          class="c8"
          src="https://i.imgur.com/7VMC0Gn.jpg"
        />
        <span>
          Just Read The Instructions 2
          <br />
          Home port: 
          Port of Los Angeles
        </span>
      </li>
      <li
        class="c7"
      >
        <img
          alt="Thumb image"
          class="c8"
          src="https://i.imgur.com/ABXtHKa.jpg"
        />
        <span>
          GO Quest
          <br />
          Home port: 
          Port Canaveral
        </span>
      </li>
    </ul>
    <div
      class="c9"
    >
      <a
        class="c10"
        href="https://youtu.be/J442-ti-Dhg"
        target="_blank"
        title="Article Link"
      >
        Article Link
      </a>
    </div>
  </div>
</main>
`
